lib/App/DuckPAN/Perl.pm

```diff
@@ -109,54 +111,73 @@ sub duckpan_install {
 		$sp =~ s/\:\:/_/g;
 
 		# special case: check for a pinned verison number
-		my $pin_version            = $ENV{$sp};
-		my $localver               = $self->get_local_version($package);
-		my $duckpan_module_version = version->parse($module->version);
-		my $duckpan_module_url     = $self->app->duckpan . 'authors/id/' . $module->distribution->pathname;
+		my $pinned_version      = $ENV{$sp};
+		my $installed_version   = $self->get_local_version($package);
+		my $latest_version      = version->parse($module->version);
+		my $duckpan_module_url  = $self->app->duckpan . 'authors/id/' . $module->distribution->pathname;
 
-		$localver ||= 1e-6 if ($pin_version); # a silly, but true, value if missing and we need to compare with pinned.
```

Comment:
> I don't believe duckpan upgrade will work in that case because my localver will be higher than anything released?

You're talking about 9.999, right?  The only time that happens is when you're using a repo for lib somehow, e.g. developing duckpan.   We could think about something like `duckpan latest` but externally that's what `upgrade` does; internally it would just allow you to circumvent the pins which I don't think we want to do.